File: Semestru 4/MPP/lab/.net/lab3/lab3/Persistence/repo/interface/ISignupRepository.cs
using Model.app.domain;

namespace Persistence.app.repo.@interface
{
    public interface ISignupRepository : IRepository<Tuple<int, int>, Signup>
    {
        IEnumerable<Signup> GetAllByChildId(int id);
        IEnumerable<Signup> GetAllByEventId(int id);
        IEnumerable<Signup> GetAllJoin();
	}
}

File: Semestru 4/MPP/lab/java/lab3/Persistence/src/main/java/org/example/repository/interf/ISignupRepository.java
package org.example.repository.interf;

import org.example.domain.Signup;
import org.example.domain.Tuple;

public interface ISignupRepository extends IRepository<Tuple<Integer, Integer>, Signup> {

    Iterable<Signup> GetAllByChildId (Integer id);
    Iterable<Signup> GetAllByEventId (Integer id);
}
